core/controllers/admin.py

```diff
@@ -1692,6 +1693,38 @@ def put(self) -> None:
         user_services.set_username(user_id, new_username)
         user_services.log_username_change(
             self.user_id, old_username, new_username)
+
+        # Update profile picture.
+        old_fs = fs_services.GcsFileSystem(
+            feconf.ENTITY_TYPE_USER, old_username)
+        new_fs = fs_services.GcsFileSystem(
+            feconf.ENTITY_TYPE_USER, new_username)
+        if old_fs.isfile('profile_picture.png'):
```

Comment:
Thinking about it we should first check if all the files exists and only then start doing modifications, so put these ifs to the above section.  line 1692

Reply:
Done